src/providers/fault-summary/cat-a-mod1/fault-summary.cat-a-mod1.ts

```diff
@@ -5,20 +5,144 @@ import { getCompetencyFaults } from '../../../shared/helpers/competency';
 export class FaultSummaryCatAM1Helper {
 
   public static getDrivingFaultsCatAM1(data: TestData): FaultSummary[] {
+    if (this.hasEmergencyStopAndAvoidanceRidingFaults(data)) {
+      return [
+        ...getCompetencyFaults(data.drivingFaults),
+        this.createEmergencyStopFaultSummary(),
```

Comment:
would create a `getSpeedCheckFaults` and implement the branching logic in there so that the `getDrivingFaultsCatAM1` function is as simple as possible